cpp/include/cudf/lists/detail/stream_compaction.hpp

```diff
@@ -34,13 +35,25 @@ std::unique_ptr<column> apply_boolean_mask(lists_column_view const& input,
                                            rmm::device_async_resource_ref mr);
 
 /**
- * @copydoc cudf::lists::distinct
+ * @copydoc cudf::lists::distinct(lists_column_view const&, null_equality, nan_equality,
+ * duplicate_keep_option, rmm::cuda_stream_view stream, rmm::device_async_resource_ref)
  */
 std::unique_ptr<column> distinct(lists_column_view const& input,
                                  null_equality nulls_equal,
                                  nan_equality nans_equal,
+                                 duplicate_keep_option keep_option,
                                  rmm::cuda_stream_view stream,
                                  rmm::device_async_resource_ref mr);
 
+/**
+ * @copydoc cudf::lists::distinct(lists_column_view const&, null_equality, nan_equality,
+ * rmm::cuda_stream_view stream, rmm::device_async_resource_ref)
+ */
+[[deprecated]] std::unique_ptr<column> distinct(lists_column_view const& input,
```

Comment:
Good job on changing the call to this function in `ColumnViewJni.cpp`, to get around the deprecation warning/error.